src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Not readable splitting of callback output calculation expression and not acceptable comparison expression splitting.

```js
const children = onlyWithSon
  ? people.filter((person) => (
      person.sex === 'm'
      && people.some((otherPerson) => (
        otherPerson.name === person.mother
      ))
    ))
  : people.filter(person => (
      people.some(otherPerson => (
        otherPerson.name === person.mother
      ))
    ));
```

Nah, even if it is more readable there is definitely and overhead in indentation. Let's extract repeated part of searching for mother in array to separate inner function.

```js
const hasMotherData = (motherName) => (
  people.some(({ name }) => name === motherName)
);

const children = onlyWithSon
  ? people.filter(({ sex, mother }) => (
      sex === 'm' && hasMotherData(mother)
    ))
  : people.filter(({ mother }) => (
      hasMotherData(mother)
    ));
```

Maybe it is worth even go further and replace filter with reduce method to add mothers data to child/son data.

```js
const findMotherData = (motherName) => (
  people.find(({ name }) => name === motherName)
);

const children = onlyWithSon
  ? people.reduce((acc, person) => {
      if (person.sex === 'f') {
        return acc;
      }
      
      const mother = findMotherData(person.mother);
      if (!mother) {
        return acc;
      }
      
      return [...acc, { ...person, mother }];
    }, [])
  : people.reduce((acc, person) => {
      const mother = findMotherData(person.mother);
      if (!mother) {
        return acc;
      }
      
      return [...acc, { ...person, mother }];
    }, []);
```